components/gitlab/sources/common/base.mjs

```diff
@@ -1,105 +1,112 @@
 import gitlab from "../../gitlab.app.mjs";
+import fetch from 'node-fetch'
+import { eventTypes } from "../common/hook-events.mjs";
 
 export default {
-  props: {
-    gitlab,
-    db: "$.service.db",
-    http: {
-      type: "$.interface.http",
-      customResponse: true,
-    },
-    projectId: {
-      propDefinition: [
+    props: {
         gitlab,
-        "projectId",
-      ],
-    },
-  },
-  hooks: {
-    /**
-     * This method should be implemented in every source component
-     */
-    async activate() {
-      // await this.activateHook(eventType);
-      throw new Error("activate() hook not implemented");
+        db: "$.service.db",
+        http: {
+            type: "$.interface.http",
+            customResponse: true,
+        },
+        projectId: {
+            propDefinition: [
+                gitlab,
+                "projectId",
+            ],
+        },
+        groupPath: {
```

Comment:
Since this prop isn't being actively used in other components, I would remove it from `base` for now (see my other comment below re: adding it into the audit event component directly).